crypto/x509/v3_utl.c

```diff
@@ -1451,3 +1463,36 @@ int ossl_bio_print_hex(BIO *out, unsigned char *buf, int len)
     OPENSSL_free(hexbuf);
     return result;
 }
+
+static int starts_with(const char *str, const char *prefix)
+{
+    return strncmp(str, prefix, strlen(prefix)) == 0;
+}
+
+static int is_valid_uri_char(char c)
+{
+    /* Valid characters include alphanumeric, '-', '_', '.', '~', and reserved characters */
+    return isalnum(c) || c == '-' || c == '_' || c == '.' || c == '~' ||
+           c == '!'|| c == '$' || c == '&' || c == '\'' || c == '(' ||
+           c == ')' || c == '*' || c == '+' || c == ',' || c == ';' ||
+           c == '=' || c == ':' || c == '@' || c == '/' || c == '#' ||
+           c == '[' || c == ']' || c == '?';
+}
+
+static int is_valid_uri(const char *uri)
+{
+    /* Check if URI begins with a valid scheme */
+    if (!(starts_with(uri, "http://") || starts_with(uri, "https://") ||
```

Comment:
mailto? file?  others?   I think only test that works is the one described at https://www.rfc-editor.org/rfc/rfc3986#page-17